src/Sulu/Bundle/WebsiteBundle/Resources/views/Sitemap/sitemap.xml.twig

```diff
@@ -4,37 +4,40 @@
     {% set registrableDomain = sulu_util_domain_info(domain).registrableDomain %}
 
     {% for entry in entries %}
-    <url>
-        <loc>{{ sulu_content_path(entry.loc, webspaceKey, locale, null, scheme) }}</loc>
-        {% if entry.lastmod %}
-        <lastmod>{{ entry.lastmod|date('Y-m-d') }}</lastmod>
-        {% endif %}
-        {% if entry.changefreq %}
-        <changefreq>{{ entry.changefreq }}</changefreq>
-        {% endif %}
-        {% if entry.priority %}
-        <priority>{{ entry.priority }}</priority>
-        {% endif %}
-
-        {% set amount = 0 %}
-        {% set defaultHref = null %}
-        {% for alternateLink in entry.alternateLinks %}
-            {% set href = sulu_content_path(alternateLink.href, webspaceKey, alternateLink.locale, null, scheme, false) %}
-            {% if href is not empty and href != alternateLink.href and sulu_util_domain_info(href).registrableDomain == registrableDomain %}
-                {% if locale != alternateLink.locale %}
-        <xhtml:link rel="alternate" hreflang="{{ alternateLink.locale|replace({'_': '-'}) }}" href="{{ href }}"/>
-                    {% set amount = amount + 1 %}
+        {% set location = sulu_content_path(entry.loc, webspaceKey, entry.locale, null, scheme) %}
+        {% if sulu_util_domain_info(location).registrableDomain == registrableDomain %}
+            <url>
+                <loc>{{ location }}</loc>
+                {% if entry.lastmod %}
+                <lastmod>{{ entry.lastmod|date('Y-m-d') }}</lastmod>
                 {% endif %}
-                {% if defaultLocale == alternateLink.locale and entry.alternateLinks|length > 1 %}
-                    {% set defaultHref = href %}
+                {% if entry.changefreq %}
+                <changefreq>{{ entry.changefreq }}</changefreq>
+                {% endif %}
+                {% if entry.priority %}
+                <priority>{{ entry.priority }}</priority>
+                {% endif %}
+
+                {% set amount = 0 %}
+                {% set defaultHref = null %}
+                {% if entry.alternateLinks|length > 1 %}
```

Comment:
maybe this would avoid to add itself as alternate so an custom provider dont need to handle it:

```twig
{% if entry.alternateLinks|length > 0 %} {# this if can maybe removed then? #}
    {% for alternateLink in entry.alternateLinks if alternateLink.locale != entry.locale %}
        {# ... #}
    {% endfor %}
{% endif %}

{% if amount > 0 %}
    <xhtml:link rel="alternate" hreflang="{{ entry.locale|replace({'_': '-'}) }}" href="{{ location }}"/>
    {% if defaultHref %}
        <xhtml:link rel="alternate" hreflang="x-default" href="{{ defaultHref }}"/>
    {% endif %}
{% endif %}
```